Equilobe.DailyReport.SL/ReportService.cs

```diff
@@ -12,27 +13,97 @@ public class ReportService : IReportService
     {
         public IJiraService JiraService { get; set; }
 
+        public JiraRequestContext JiraRequestContext { get; set; }
+
         #region IReportService Implementation
         public void UpdateDashboardData(long instanceId)
         {
-            UpdateAtlassianUsers(instanceId);
+            JiraRequestContext = GetJiraRequestContext(instanceId);
+
+            SyncAtlassianUsers(instanceId);
+            SyncAtlassianWorklogs(instanceId);
         }
         #endregion
 
         #region Helpers
-        private void UpdateAtlassianUsers(long instanceId)
+        private void SyncAtlassianWorklogs(long instanceId)
         {
-            var jiraRequestContext = GetJiraRequestContext(instanceId);
-            var users = JiraService.GetAllUsers(jiraRequestContext)
-                .Select(p => ToAtlassianUser(p, instanceId))
-                .ToList();
+            SyncDeletedWorklogs(instanceId);
+            SyncUpdatedWorklogs(instanceId);
+        }
+
+        private void SyncDeletedWorklogs(long instanceId)
+        {
+            var deletedWorklogsIds = JiraService.GetDeletedWorklogsIds(JiraRequestContext, DateTime.UtcNow.AddMonths(-7));
+
+            if (deletedWorklogsIds == null)
+                return;
 
             using (var db = new ReportsDb())
             {
-                var dbUsers = db.AtlassianUser
+                var dbWorklogs = db.AtlassianWorklog
                     .Where(p => p.InstalledInstanceId == instanceId)
                     .ToList();
 
+                foreach (var worklogId in deletedWorklogsIds)
+                {
+                    var dbWorklog = dbWorklogs.SingleOrDefault(p => p.JiraWorklogId == worklogId);
+
+                    if (dbWorklog != null)
+                        db.AtlassianWorklog.Remove(dbWorklog);
```

Comment:
Update the collection name on db context also to be plural.

Reply:
Shouldn't it be on singular? AtlassianWorklog is a database table, not a collection and I find it normal to be in a singular state, because it's a db table which contains multiple entries of an atlassian worklog

Reply:
This is the dbSet. And it should also be plural. All dbSets in our app are plural. That was the convention. If you put this on singular, it will break the consistency. 
Also, this is the code in c#, not necessary the table. You can have the pluralization in code, and the table in singular (but right now, we also have the table pluralized)